src/layer/AbstractSentinelHubV1OrV2Layer.ts

```diff
@@ -140,4 +155,54 @@ export class AbstractSentinelHubV1OrV2Layer extends AbstractLayer {
 
     return response.data.map((date: string) => moment.utc(date).toDate());
   }
+
+  public async getStats(params: GetStatsParams): Promise<GetStats> {
+    if (!params.geometry) {
+      throw new Error('Parameter "geometry" needs to be provided');
+    }
+    if (!params.crs) {
+      throw new Error('Parameter "crs" needs to be provided');
+    }
+    if (!params.resolution) {
+      throw new Error('Parameter "resolution" needs to be provided');
+    }
+    if (!params.fromTime || !params.toTime) {
+      throw new Error('Parameters "fromTime" and "toTime" needs to be provided');
+    }
+
+    const payload: FisPayload = {
+      layer: this.layerId,
+      crs: params.crs.authId,
+      geometry: WKT.convert(params.geometry),
+      time: `${moment.utc(params.fromTime).format('YYYY-MM-DDTHH:mm:ss') + 'Z'}/${moment
+        .utc(params.toTime)
+        .format('YYYY-MM-DDTHH:mm:ss') + 'Z'}`,
+      resolution: undefined,
+      bins: params.bins || 5,
+      type: HistogramType.EQUALFREQUENCY,
+      ...this.getStatsAdditionalParameters(),
+    };
+    // When using CRS=EPSG:4326 one has to add the "m" suffix to enforce resolution in meters per pixel
+    if (params.crs.authId === CRS_EPSG4326.authId) {
+      payload.resolution = params.resolution + 'm';
+    } else {
+      payload.resolution = params.resolution;
+    }
+    if (this.evalscript) {
+      payload.evalscript = btoa(this.evalscript);
+      payload.evalsource = this.getEvalsource();
+    }
+
+    const { data } = await axios.get(this.dataset.shServiceHostname + 'v1/fis/' + this.instanceId, {
+      params: payload,
+    });
+    // convert date strings to Date objects
+    for (let channel in data) {
+      data[channel] = data[channel].map((dailyStats: StatsPerChannel) => ({
```

Comment:
Sorry, I know I suggested this, but it's actually wrong. The input `dailyStats` is not of `StatsPerChannel` type (it has a string instead of a Date object). I would suggest:
```
data[channel] = data[channel].map((dailyStats: any) => ({
  ...
```

Reply:
True, but this makes me think that `StatsPerChannel` is wrongly named and would suggest `StatsPerDay`?

Reply:
Also true, yes... Though it's actually `StatsPerChannelPerDay`... :) Maybe `DailyChannelStats`?

Reply:
Changed to `DailyChannelStats`